scene/3d/remote_transform.cpp

```diff
@@ -0,0 +1,132 @@
+
+/*************************************************************************/
+/*  remote_transform.cpp                                              */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                    http://www.godotengine.org                         */
+/*************************************************************************/
+/* Copyright (c) 2007-2016 Juan Linietsky, Ariel Manzur.                 */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "remote_transform.h"
+
+void RemoteTransform::_update_cache() {
+	cache=0;
+	if (has_node(remote_node)) {
+		Node *node = get_node(remote_node);
+		if (!node || this==node || node->is_a_parent_of(this) || this->is_a_parent_of(node)) {
+			return;
+		}
+
+		cache=node->get_instance_ID();
+	}
+}
+
+void RemoteTransform::_update_remote() {
+
+
+	if (!is_inside_tree())
+		return;
+
+	if (!cache)
+		return;
+
+	Object *obj = ObjectDB::get_instance(cache);
+	if (!obj)
+		return;
+
+	Spatial *n = obj->cast_to<Spatial>();
+	if (!n)
+		return;
+
+	if (!n->is_inside_tree())
+		return;
+
+	//todo make faster
```

Comment:
@bojidar-bg have no idea how this can be faster, this comment is in Remote2DTransform node as well, was unsure if I should delete it, in the end I didn't